doc/source/user-environment.rst

```diff
@@ -200,11 +200,37 @@ user interface for Project Jupyter. It can be used with JupyterHub, both as an
 optional interface and as a default.
 
 1. `Install JupyterLab <https://github.com/jupyterlab/jupyterlab#installation>`_
-   in your user image.
+   in your user image, for example in your Dockerfile:
+
+   .. code-block::
+
+      FROM jupyter/base-notebook:27ba57364579
+
+      ...
+      ARG JUPYTERLAB_VERSION=0.31.10
+      RUN pip install jupyterlab==$JUPYTERLAB_VERSION
+      ...
+
 2. `Install JupyterLab Hub extension
    <https://github.com/jupyterhub/jupyterlab-hub#installation>`_ in your user
-   image. This provides a nice UI for accessing JupyterHub control panel from
-   JupyterLab. You only need the ``jupyter labextension`` command.
+   image:
+
+   .. code-block:: bash
+
+      $ jupyter labextension install @jupyterlab/hub-extension
```

Comment:
This should probably be in "Dockerfile" form too, since the other two instructions in this list are as well. (I assume you ran this in the Dockerfile, yeah?)